### PR TITLE
Allow setting Source of a Devolve

### PR DESCRIPTION
**Pull request by flagrama**
Allows fixing bugs like https://forum.tcgone.net/t/bug-report-bent-spoon-fco-93-espeon-exs-miraculous-s/2867

### CODE REVIEW

File: src/tcgwars/logic/groovy/TcgStatics.groovy [hunk 1]
```diff
@@ -611,10 +611,14 @@ class TcgStatics {
...
   }
-  static devolve (PokemonCardSet pcs, Card card, CardList newLocation){
-    def blocked = bg().em().run(new MoveCard(card, newLocation));
-    if (blocked) {
-      return;
+  static devolve (PokemonCardSet pcs, Card card, CardList newLocation, Source source=ATTACK){
+    def blockedDevolve = bg().em().run(new Devolve(pcs, source))
+    if (blockedDevolve) {
+      return
+    }
```
Comment (axpendix) [inline src/tcgwars/logic/groovy/TcgStatics.groovy:618]: `Devolve` runs `CheckAbilities`. By the time it runs, the card has to be moved away, but this change prevents that, thus the abilities will be incorrect after this change.

**Comment by axpendix** [comment]
@flagrama any updates on this?

**Comment by flagrama** [comment]
Not yet, though I'm not sure exactly how to handle it either since either the card move or the devolve can be blocked, and if the card gets moved but the devolve gets blocked it can trigger effects that shouldn't be triggered at that time. Will probably look into moving it to a temporary card list, then devolve, then move to the desired location.

**Comment by axpendix** [comment]
This PR is irrelevant thus closed. `Devolve` has already been refactored as a full effect and I'm working on source refactoring to fix sources of all use cases at the core.

**Pull request closed without merging**